### PR TITLE
Upload map files to specific artifact locations

### PR DESCRIPTION
**Pull request by danieljurek**
Eliminates the problem of colliding artifacts on retry

### CODE REVIEW

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 1]
```diff
@@ -202,9 +202,10 @@ jobs:
           condition: and(succeededOrFailed(), eq(variables['CODE_COVERAGE'], 'enabled'), eq(variables['Skip.BranchCoverageEnforcement'], ''))
 
       # Use the job name to create the artifact name for MAP file publishing.
-      # "Validate Windows2019_UWP_debug_x86" -> "Windows2019_UWP_debug_x86"
+      # Attempts are also noted starting with 1
+      # "Validate Windows2019_UWP_debug_x86" -> "Windows2019_UWP_debug_x86_attempt_1"
       - pwsh: |
-          $artifactName = '$(Agent.JobName)'
+          $artifactName = "$(Agent.JobName)_attempt_$(Agent.JobAttempt)"
```
Comment (ahsonkhan) [inline eng/pipelines/templates/jobs/ci.tests.yml:208]: When do we use single quote vs double quote?

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 2]
```diff
@@ -202,9 +202,10 @@ jobs:
           condition: and(succeededOrFailed(), eq(variables['CODE_COVERAGE'], 'enabled'), eq(variables['Skip.BranchCoverageEnforcement'], ''))
 
       # Use the job name to create the artifact name for MAP file publishing.
-      # "Validate Windows2019_UWP_debug_x86" -> "Windows2019_UWP_debug_x86"
+      # Attempts are also noted starting with 1
```
Comment (ahsonkhan) [inline eng/pipelines/templates/jobs/ci.tests.yml:205]: 👍

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 3]
```diff
@@ -219,9 +220,11 @@ jobs:
...
+        displayName: Stage map files
 
       - task: PublishPipelineArtifact@1
         inputs:
           artifactName: map-files-$(MapFileArtifactName)
           path: $(Build.ArtifactStagingDirectory)
-        condition: eq(variables['PublishMapFiles'], 'true')
\ No newline at end of file
+        condition: eq(variables['PublishMapFiles'], 'true')
+        displayName : Publish map file artifacts
```
Comment (ahsonkhan) [inline eng/pipelines/templates/jobs/ci.tests.yml:230]: nit: eof doesn't have a new line

**Review by ahsonkhan — APPROVED** [review]

**Comment by ahsonkhan** [comment]
Btw, is there a way to test that a failed re-try attempt will work now?

**Comment by danieljurek** [comment]
> Btw, is there a way to test that a failed re-try attempt will work now?

Not easily in the current system. One way this could be accomplished is by adding a fault injection build variable and a step to fail the pipeline if that build variable is set... but you'd have to put it in the correct place in the pipeline which will change depending on what you're trying to test.. You'd end up modifying the code almost every time you want to do fault injection anyway.

**Review by vhvb1989 — APPROVED** [review]
<3

**Pull request merged**